frontend/src/components/AddGame/AddGame.tsx

```diff
@@ -38,93 +85,124 @@ const AddGame: FC<AddGameProps> = () => {
 	}
 
 	return (
-		<form
-			onSubmit={(e) => {
-				e.preventDefault();
-				postData({
-					name,
-					description,
-					platform,
-					releaseDate: releaseDate?.toISOString(),
-					playtime,
-					playerMin,
-					playerMax,
-					location
-				});
-			}}
-		>
-			<TextInput
-				label="Name of the game"
-				type="text"
-				onChange={(input) => setName(input.currentTarget.value)}
-				value={name}
-			/>
-			<br />
-
-			<TextArea
-				label="Description fo the game"
-				onChange={(textarea) => setDescription(textarea.currentTarget.value)}
-				value={description}
-			/>
-			<br />
-
-			<Select
-				label="Platform"
-				options={data.map((platform) => platform.name)}
-				placeholder="Select a platform"
-				onChange={(select) => setPlatform(select.target.value)}
-				value={platform}
-			/>
-			<br />
-
-			<DateInput
-				label="Release date"
-				onChange={(input) =>
-					setReleaseDate(new Date(input.currentTarget.value))
-				}
-				value={releaseDate?.toISOString().split('T')[0] || ''}
-			/>
-			<br />
-
-			<TextInput
-				label="Expected playtime"
-				type="number"
-				onChange={(input) =>
-					setPlaytime(Number.parseInt(input.currentTarget.value))
-				}
-				value={playtime?.toString() || ''}
-			/>
-			<br />
-			<TextInput
-				label="Minimum number of players"
-				type="number"
-				onChange={(input) =>
-					setPlayerMin(Number.parseInt(input.currentTarget.value))
-				}
-				value={playerMin?.toString() || ''}
-			/>
-			<br />
-			<TextInput
-				label="Maximum number of players"
-				type="number"
-				onChange={(input) =>
-					setPlayerMax(Number.parseInt(input.currentTarget.value))
-				}
-				value={playerMax?.toString() || ''}
-			/>
-			<br />
-			<TextInput
-				label="Location of the game"
-				type="text"
-				onChange={(input) =>
-					setLocation(input.currentTarget.value)
-				}
-				value={location}
-			/>
-			<br />
-
-			<input type="submit" value="Submit" />
-		</form>
+		<>
+			<form
+				onSubmit={handleSubmit(
+					async (d) => {
+						try {
+							await toast.promise(
+								postDataAsync(d),
+								{
+									pending: 'Adding game...',
+									success: 'Game added!',
+									error: {
+										render: () => {
+											return (
+												<>
+													{postError?.response?.status === 401
+														? 'You need to be signed in to add games'
+														: 'Something went wrong!'}
+												</>
+											);
+										}
+									}
+								},
+								{
+									position: 'bottom-right'
+								}
+							);
+							resetForm();
+						} catch (e) {}
+					},
+					(e) =>
+						toast.error('Please fill out all fields correctly!', {
+							position: 'bottom-right'
+						})
+				)}
```

Comment:
Please move this to separate function, not inline

Reply:
Better now?